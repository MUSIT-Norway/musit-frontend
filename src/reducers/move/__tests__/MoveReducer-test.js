import deepFreeze from 'deep-freeze';
import MuseumId from '../../../shared/models/museumId';
import * as actions from '../index';
import Config from '../../../config';
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);

const sendToBackend = [
  {
    doneBy: 3,
    items: [1],
    destination: 1
  }
];

deepFreeze(sendToBackend);

describe('MoveReducer', () => {
  it('creates MOVE_OBJECT_SUCCESS when fetching data has been done', () => {
    const putData = {
      doneBy: 'f4e1eba9-8add-4675-9836-db59f7115d7c',
      destination: 4,
      items: [2, 3]
    };
    const url = `${Config.magasin.urls.storagefacility.baseUrl(new MuseumId(99))}/moveObject`;
    nock('http://localhost')
      .put(url, putData)
      .reply(201, {
        moved: [2, 3],
        failed: []
      });
    const store = mockStore();

    return store.dispatch(actions.moveObject([2, 3], 4, 'f4e1eba9-8add-4675-9836-db59f7115d7c', new MuseumId(99)))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates MOVE_NODE_SUCCESS when fetching data has been done', () => {
    const putData = {
      doneBy: 1,
      destination: 4,
      items: [2, 3]
    };
    const url = `${Config.magasin.urls.storagefacility.baseUrl(new MuseumId(99))}/moveNode`;
    nock('http://localhost')
      .put(url, putData)
      .reply(201, {
        moved: [2, 3],
        failed: []
      });
    const store = mockStore();

    return store.dispatch(actions.moveNode([2, 3], 4, 1, new MuseumId(99)))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
