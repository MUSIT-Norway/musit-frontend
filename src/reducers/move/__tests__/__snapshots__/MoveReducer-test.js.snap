exports[`MoveReducer creates MOVE_NODE_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/move/node/start"
  },
  Object {
    "result": Object {
      "failed": Array [],
      "moved": Array [
        2,
        3
      ]
    },
    "type": "musit/move/node/success"
  }
]
`;

exports[`MoveReducer creates MOVE_OBJECT_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/move/object/start"
  },
  Object {
    "result": Object {
      "failed": Array [],
      "moved": Array [
        2,
        3
      ]
    },
    "type": "musit/move/object/success"
  }
]
`;

exports[`MoveReducer node: fail action 1`] = `
Object {
  "error": [Error: Some error in moveNode.],
  "loaded": false,
  "loading": false
}
`;

exports[`MoveReducer node: initial action 1`] = `
Object {
  "loaded": false,
  "loading": true
}
`;

exports[`MoveReducer node: no action 1`] = `Object {}`;

exports[`MoveReducer node: success action 1`] = `
Object {
  "loaded": true,
  "loading": false
}
`;

exports[`MoveReducer object: fail action 1`] = `
Object {
  "error": [Error: Some error in moveObject.],
  "loaded": false,
  "loading": false
}
`;

exports[`MoveReducer object: initial action 1`] = `
Object {
  "loaded": false,
  "loading": true
}
`;

exports[`MoveReducer object: no action 1`] = `Object {}`;

exports[`MoveReducer object: success action 1`] = `
Object {
  "loaded": true,
  "loading": false
}
`;
